ee/codegen/src/generators/nodes/search-node.ts

```diff
@@ -259,39 +272,32 @@ export class SearchNode extends BaseSingleFileNode<
     return statements;
   }
 
-  private generateInputVariableIdsByLogicalIdMap(
+  private generateMetadataFilterInputIdByOperandIdMap(
     rawData: VellumLogicalExpression
   ): Map<string, string> {
     const result = new Map<string, string>();
-    const prefix = "vellum-query-builder-variable-";
 
     const traverse = (logicalExpression: VellumLogicalExpression) => {
       if (logicalExpression.type === "LOGICAL_CONDITION") {
-        const lhsQueryInput = this.nodeInputsByKey.get(
-          `${prefix}${logicalExpression.lhsVariableId}`
-        )?.nodeInputData?.value.rules[0] as InputVariablePointer;
-        const rhsQueryInput = this.nodeInputsByKey.get(
-          `${prefix}${logicalExpression.rhsVariableId}`
-        )?.nodeInputData?.value.rules[0] as InputVariablePointer;
+        const lhsQueryInput = this.nodeInputsById.get(
+          logicalExpression.lhsVariableId
+        )?.nodeInputData?.id;
+        const rhsQueryInput = this.nodeInputsById.get(
+          logicalExpression.rhsVariableId
+        )?.nodeInputData?.id;
```

Comment:
Perform lookups by id rather than by key (which required deriving the key the same way the main repo does and duplicating logic across the two)